samples/dashboards/demo/personal-portfolio/demo.js

```diff
@@ -0,0 +1,429 @@
+const basicInvestmentPlan = {
+    interval: 30, // Every 30 days
+    amount: 200   // Amount in EUR
+};
+
+
+const stockCollection = [
+    {
+        tradingSymbol: 'NFLX',
+        ISIN: 'US64110L1061',
+        SecID: '0P000003UP'
+    }, {
+        tradingSymbol: 'MSFT',
+        ISIN: 'US5949181045',
+        SecID: '0P000003MH'
+    }, {
+        tradingSymbol: 'AMZN',
+        ISIN: 'US0231351067',
+        SecID: '0P000000B7'
+    }, {
+        tradingSymbol: 'GOOGL',
+        ISIN: 'US02079K3059',
+        SecID: '0P000002HD'
+    }
+];
+
+
+const generatePortfolio = (investmentPlan, stockPrices) => {
+    const { interval, amount } = investmentPlan,
+        portfolioValues = [];
+    let totalUnits = 0;
+    stockPrices.forEach((priceData, day) => {
+        // Check if it's an investment day
+        if ((day % interval) === 0) {
+            totalUnits += amount / priceData[1];
+        }
+
+        // Calculate portfolio value for the day
+        const value = totalUnits * priceData[1];
+        portfolioValues.push([priceData[0], value]);
+    });
+    return portfolioValues;
+};
```

Comment:
Could this be replaced by a MathModifier? @ddragula @karolkolodziej Perhaps by creating a column in the table with `amount`. I think we should aim to reduce the amount of code.

Reply:
Thanks for investigating this!